main.go

```diff
@@ -45,8 +45,13 @@ func main() {
 		os.Exit(0)
 	}
 
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create secretmanager client: %v", err)
+	}
+
 	// Fetch and write secrets.
-	if err := plugin(ctx); err != nil {
+	if err := plugin(ctx, client, *attributes, *secrets, *targetPath, *permission); err != nil {
```

Comment:
I have a general rule that when a function's arity exceeds 4, it should accept a struct instead:

```go
plugin(ctx, &pluginInput{
   Attributes: *attributes,
   // ...
})
```

Reply:
done